### PR TITLE
chore: Bump golang and tools versions and dependencies

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

Bump to golang 1.21, the latest tool versions and bump the dependencies where possible.

Changes are made to the code to allow usage of the latest dependencies and to satisfy golangci-lint.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -88,10 +90,8 @@ linters:
     - stylecheck
     - typecheck
     - unconvert
-    - unparam
     - unused
     - whitespace
-    - ginkgolinter
```
Comment (jcanocan) [inline .golangci.yml:94]: Why we need to get rid of the `ginkgolinter`?

File: Makefile [hunk 2]
```diff
@@ -12,22 +12,22 @@ medius:
 	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/medius kubevirt.io/containerdisks/cmd/medius
 
 fmt:
-	go mod tidy -compat=1.19
+	go mod tidy -compat=1.21
```
Comment (jcanocan) [inline Makefile:15]: Isn't required to specify the Z version?

**Review by jcanocan — COMMENTED** [review]
Nice job! Just a couple of silly questions :)

File: Makefile (hunk 2 above)
Comment (0xFelix) [reply Makefile:15]: Not here.

File: .golangci.yml (hunk 1 above)
Comment (jcanocan) [reply .golangci.yml:94]: All right! Thanks for the explanation.

**Pull request merged**